### ISSUE TITLE
SF V2 - some HTML entities being incorrectly escaped during push

### ISSUE THREAD

**Issue by rhobkirkcertinia** [forcedotcom/cli#2448]
### Summary
In our `customlabels` we have some HTML entities that get pushed up to the org.

These are things like `&lt;` , `&gt;` and `&#160;` etc. All of these seem to work except `&#160;` which gets encoded when pushing to the org and ends up rendering as the literal character code when used.

`&#160;` is a non-breaking space and is the same as using `&nbsp;`. We've tried switching over to that and it also got encoded and therefore didn’t work either. This appears to be a regression as SFDX cli version 7.183.1 doesn’t have this issue.

This issue might exist with other characters too - this is the only one that we've spotted so far.

### Steps To Reproduce
1. Create a label similar to this 
    ```
    <labels>
        <fullName>example</fullName>
        <categories>example</categories>
        <language>en_US</language>
        <protected>true</protected>
        <shortDescription>example.</shortDescription>
        <value>3 results,&#160;and 1 other.</value>
    </labels>
    ```
1. Import that label into an LWC and render it out on the HTML and add it to a page.
1. Deploy that code to an org using SFDX 7.183.1
1. View the component in an org and view the LWC and it should show the label as `3 results, and 1 other`
1. Change to SF version 2.5.8
1. Deploy the same code to the org again (or a new org)
1. Now it displays as `3 results,&#160;and 1 other`

### Expected result
HTML entities are not escaped so that they are rendered as expected (and per previous CLI behaviour) 

### Actual result
HTML entities are escaped, causing them to be rendered as the string literal version of the character

### System Information
@salesforce/cli/2.5.8 darwin-arm64 node-v18.15.0

**Comment by mshanemc** [comment]
found it, got a fix PR ready.

**Comment by rhobkirkcertinia** [comment]
Brilliant, thanks @mshanemc!

**Comment by mshanemc** [comment]
@rhobkirkcertinia can you try this with your codebase using `nightly` for `sf` to make sure this solved all of them.

**Comment by mshanemc** [comment]
welp, that turned out really bad (caused a regression with the parsing of some other html entities).  https://github.com/forcedotcom/cli/issues/2455

gonna roll that back and come up with some other opton.

**Comment by rhobkirkcertinia** [comment]
Ouch, thanks for trying @mshanemc . I'll hang fire testing until there is a new solution - please can you re-open this issue in the mean time?

**Comment by mshanemc** [comment]
this issue stayed open, only the PR associated was closed

**Comment by mshanemc** [comment]
OK, after a much better investigation than the first time, here's what I've found.

### sfdx < 7.184

Local project `value>3 results,&#160;and 1 other.</value>` 
CLI would deploy labels with html markup "as is"
CustomLabel in the CustomLabels UI would show the "interpreted" value `3 results, and 1 other.` but if you inspect there (or in an LWC) you'd see the html characters from the label.
If you retrieve from the org, you'd get `<value>3 results, and 1 other.</value>` into the local project.  This happens even in mdapi format--the special characters are lost and converted to a "regular space"

### since then (sfdx >= 7.184 and sf >=2 )

CLI was escaping the ampersand character `&` so that `value>3 results,&#160;and 1 other.</value>`  would deploy, 
Showed up in CustomLabels UI and LWC as 3 results,&amp;#160;and 1 other.
retrieve as `value>3 results,&amp;#160;and 1 other.</value>` which is how it was showing up in the LWC.

I think I can fix that safely.  It'll still clobber labels down to a ` ` (space) when you retrieve it.

**Comment by rhobkirkcertinia** [comment]
Thanks @mshanemc 👍

**Comment by iowillhoit** [comment]
Fixed in the latest version of `sf` `2.12.9`
